datafusion/functions-window/src/rank.rs

```diff
@@ -80,11 +125,19 @@ impl WindowUDFImpl for Rank {
         &self,
         _partition_evaluator_args: PartitionEvaluatorArgs,
     ) -> Result<Box<dyn PartitionEvaluator>> {
-        Ok(Box::<RankEvaluator>::default())
+        Ok(Box::new(RankEvaluator {
+            state: RankState::default(),
+            rank_type: self.rank_type,
+        }))
     }
 
     fn field(&self, field_args: WindowUDFFieldArgs) -> Result<Field> {
-        Ok(Field::new(field_args.name(), DataType::UInt64, false))
+        let nullable = false;
+        Ok(Field::new(
+            field_args.name(),
+            self.data_type.clone(),
+            nullable,
+        ))
```

Comment:
The `Rank.data_type` field can be removed.


```suggestion
        let return_type = match self.rank_type {
            RankType::Basic | RankType::Dense => DataType::UInt64,
            RankType::Percent => DataType::Float64,
        };
        
        Ok(Field::new(
            field_args.name(),
            return_type.clone(),
            false,
        ))
```

Reply:
```
impl Rank {
    pub fn new(name: String, rank_type: RankType) -> Self {
        let data_type = if matches!(rank_type, RankType::Percent) {
            DataType::Float64
        } else {
            DataType::UInt64
        };

        Self {
            name,
            signature: Signature::any(0, Volatility::Immutable),
            rank_type,
            data_type,
        }
    }


```

This logic has been addressed during the initialization of the Rank struct.


Should I still make this change ? 
@jcsherin